agent/check_handler.go

```diff
@@ -227,7 +227,7 @@ func (a *Agent) executeCheck(ctx context.Context, request *corev2.CheckRequest,
 	event.Timestamp = time.Now().Unix()
 
 	if len(checkHooks) != 0 {
-		event.Check.Hooks = a.ExecuteHooks(request, checkExec.Status)
+		event.Check.Hooks = a.ExecuteHooks(request, checkExec.Status, assets)
```

Comment:
I'm not very familiar with this code, but at a glance it seems that `assets` here corresponds to the check's assets? Is it not possible for checks and hooks to use different assets?

Reply:
Yes, it's completely possible for checks and hooks to use different assets. They're all [loaded in by schedulerd](https://github.com/sensu/sensu-go/blob/feature/hook-assets/backend/schedulerd/executor.go#L347-L366) and send to the agent. I should update the variable name here to distinguish it is not just the check's assets, rather the entire request.

Reply:
I see, the scheduler sends the agent a list of all the assets that are needed for it to be able to run the check, its hooks, ... all in one go, as one big bundle? I assumed it was "compartmentalized" but never checked.

My question then is as follow: could there be situations in which the check's assets and the hook's assets are fine as separate sets, but incompatible when merged? I'm not expecting this PR to attempt to solve such issues, but I'm curious to know what you think about it!

Reply:
For each check being scheduled, a request is built with the check, it's hooks, and it's assets (this includes all assets needed by the check and the potential hooks). You are correct, the check request is sent from the backend to the agent as one big bundle. I'm not sure how to compartmentalize this to be more efficient, since only the backend is aware of assets which exist.

I'm not sure what you mean by "incompatible when merged." Could you elaborate?

Reply:
The compartmentalization I mention is more about guaranteeing the environment checks and hooks expect more than anything else. I know our assets are pretty rudimentary, but let's take typical package manager problems into consideration. Suppose checkAsset is an asset used by a check and hookAsset is an asset used by a hook.

- checkAsset and hookAsset are not compatible

- checkAsset depends on A at version X; hookAsset depends on A at version Y
After all assets are installed, the check or the hook might not work

- checkAsset includes a dependency on A; hookAsset includes a dependency on B; A and B are incompatible
After all assets are installed, the check and the hook might not work.

In each of those cases, "running" {check + its assets} or {hook + its assets} would work, but "running" {check + all assets} or {hook + all assets} might not (or worse, produce wrong results).

I need to check how assets are used by the agent; either way, this is not an issue with your PR at all, but that's what I meant by "incompatible when merged"!

Reply:
Hm, I see. If there were assets that were incompatible, they would cached in different directories (as all assets are), and only the environments of the relevant assets are loaded into the executable's environment. I suppose that is sufficient for now, but we can definitely revisit this if we run into an issue in the future!